tests/db_functions/json/test_json_array.py

```diff
@@ -107,6 +107,45 @@ def test_order_by_nested_key(self):
         )
         self.assertQuerySetEqual(qs, Author.objects.order_by("-alias"))
 
+    def test_null_on_null(self):
+        obj = Author.objects.annotate(
+            arr=JSONArray(F("goes_by"), absent_on_null=False)
+        ).first()
+
+        self.assertEqual(obj.arr, [None])
+
+    @skipIfDBFeature("supports_json_absent_on_null")
+    def test_absent_on_null_not_supported(self):
+        msg = "ABSENT ON NULL is not supported by this database backend."
+        with self.assertRaisesMessage(NotSupportedError, msg):
+            Author.objects.annotate(
+                arr=JSONArray(F("goes_by"), absent_on_null=True)
+            ).first()
+
+    @skipUnlessDBFeature("supports_json_absent_on_null")
+    def test_absent_on_null(self):
+        obj = Author.objects.annotate(
+            arr=JSONArray(F("name"), F("goes_by"), absent_on_null=True)
+        ).first()
+        self.assertEqual(obj.arr, ["Ivan Ivanov"])
```

Comment:
The following should ensure the null eliding takes place at the database level

```suggestion
        obj = Author.objects.annotate(
            arr=JSONArray(F("goes_by"), F("name"), absent_on_null=True),
            arr_first=F("arr__0"),
        ).first()
        self.assertEqual(obj.arr, ["Ivan Ivanov"])
        self.assertEqual(obj.arr_first, "Ivan Ivanov")
```

Reply:
That's not a bad idea, and actually b y using the F expression, you're also testing that you can have the database perform operations on the array on the database side.

Should I apply the same technique to `test_single_absent_on_null` as well? I'm not entirely sure the value in even having the two tests in `test_single_absent_on_null`. Can we just drop that test?

Reply:
I think it might be worth keeping `test_empty_absent_on_null` as is if only to make sure there's no backend that do the crazy thing of returning `None` when there's no members at all.

Reply:
Also reading over all of the JSON array tests, it's possible we might be missing something with nested JSONArray with differing absent on null.

For example, I would hope something like `JSONArray(Value(1), Value(2), JSONArray(None), absent_on_null=True)` would result in `[1, 2, [None]]` and not `[1, 2, []]` or `[1, 2]`

Reply:
Good catch, might be worth adding a test for nested array just to make sure.

Reply:
I don't think it's necessary to test all four possible combinations of inner/outer absent_on_null, but I think just adding something equivalent to this case might be valuable

Reply:
I'll get it going.

Reply:
Hey, this actually revealed an error.

```
======================================================================
FAIL: test_nested_absent_on_null (db_functions.json.test_json_array.JSONArrayTests.test_nested_absent_on_null)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/john/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/lib/python3.13/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/home/john/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/lib/python3.13/unittest/case.py", line 651, in run
    self._callTestMethod(testMethod)
    
  File "/home/john/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/lib/python3.13/unittest/case.py", line 606, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^
  File "/home/john/Code/django/django/test/testcases.py", line 1550, in skip_wrapper
    return test_func(*args, **kwargs)
    ^^^^^^^
  File "/home/john/Code/django/tests/db_functions/json/test_json_array.py", line 162, in test_nested_absent_on_null
    self.assertEqual(obj.arr, ["Ivan Ivanov", ["iivanov", None]])
    ^^^^^^^^^^^^^^^
  File "/home/john/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/lib/python3.13/unittest/case.py", line 907, in assertEqual
    assertion_func(first, second, msg=msg)
    ^^^^^^^^^^^^^^^
  File "/home/john/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/lib/python3.13/unittest/case.py", line 1113, in assertListEqual
    self.assertSequenceEqual(list1, list2, msg, seq_type=list)
    ^^^^^^^^^^^^^^^
  File "/home/john/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/lib/python3.13/unittest/case.py", line 1095, in assertSequenceEqual
    self.fail(msg)
    ~~~~~~~~~^^^^^
  File "/home/john/.local/share/uv/python/cpython-3.13.0-linux-x86_64-gnu/lib/python3.13/unittest/case.py", line 732, in fail
    raise self.failureException(msg)
    ^^^^^^^^^^^^^^^
AssertionError: Lists differ: ['Ivan Ivanov', '["iivanov",null]'] != ['Ivan Ivanov', ['iivanov', None]]

First differing element 1:
'["iivanov",null]'
['iivanov', None]

- ['Ivan Ivanov', '["iivanov",null]']
?                  --       ^  ^^^ -

+ ['Ivan Ivanov', ['iivanov', None]]
?                 +        ^ +++ ^


----------------------------------------------------------------------
```

Here's the test for reference

```python
  @skipUnlessDBFeature("supports_json_absent_on_null")
  def test_nested_absent_on_null(self):
      obj = Author.objects.annotate(
          arr=JSONArray(
              F("name"),
              JSONArray(F("alias"), F("goes_by")),
              F("goes_by"),
              absent_on_null=True,
          )
      ).first()
      self.assertEqual(obj.arr, ["Ivan Ivanov", ["iivanov", None]])
```

So it looks like there's a serious issue with the custom function that it doesn't properly handle nested arrays.

Reply:
I suspect you won't be able to get the above test to pass on backends that don't have a native JSON type even when reverting the function. It's the case for SQLite where data is passed around as JSON serialized string.

Reply:
For `JSONObject` on SQLite you could use [the `json_patch('{}', json_object(...))` trick](https://sqlite.org/json1.html#jpatch) as it elides null

```sql
sqlite> SELECT json_patch('{}', json_object("dob", NULL, "name", "foo"));
{"name":"foo"}
```